03_Modules/OcppRouter/src/module/api.ts

```diff
@@ -55,16 +58,35 @@ export class AdminApi
    * @param {FastifyRequest<{ Body: Subscription }>} request - The request object, containing the body which is parsed as a {@link Subscription}.
    * @return {Promise<number>} The id of the created subscription.
    */
-  @AsDataEndpoint(Namespace.Subscription, HttpMethod.Post)
+  @AsDataEndpoint(
+    Namespace.Subscription,
+    HttpMethod.Post,
+    undefined,
+    CreateSubscriptionSchema,
+  )
   async postSubscription(
     request: FastifyRequest<{ Body: Subscription }>,
   ): Promise<number> {
+    if (
+      !request.body.onClose &&
+      !request.body.onConnect &&
+      !request.body.onMessage &&
+      !request.body.sentMessage
+    ) {
+      throw new Error(
```

Comment:
Nitpick: shouldn't we be returning a HTTP 400 (instead of a 500) in this case? The client is sending us something that isn't allowed.

Reply:
It looks like below now,
<img width="1002" alt="Screenshot 2024-07-24 at 11 10 38 AM" src="https://github.com/user-attachments/assets/18cc60d1-c4bd-4593-b597-fd138c316c25">